emitter/src/tests/action.spec.ts

```diff
@@ -225,4 +242,28 @@ describe('Actions', () => {
     fixture.componentInstance.decrement!.emit();
     fixture.componentInstance.multiplyBy2!.emit();
   });
+
+  it('should dispatch actions to correct receivers', () => {
+    TestBed.configureTestingModule({
+      imports: [NgxsModule.forRoot([CounterState]), NgxsEmitPluginModule.forRoot()],
+      declarations: [MockComponent]
+    });
+
+    const store: Store = TestBed.get(Store);
+    const actions$: Actions = TestBed.get(Actions);
+
+    const action = actions$
+      .pipe(
+        ofEmittableSuccessful(CounterState.dontChange, CounterState.increment, CounterState.decrement),
+        take(3)
+      )
+      .toPromise()
```

Comment:
This shouldn't be asynchronous test, let's stick with synchronous approach.

Reply:
I need to wait for actions to be completed before inspecting state and do assertions. I don't know any approaches to write that without using asynchronous test.

Reply:
The code inside this test is synchronous, there is any micro/macro tasks running, so you don't need a Promise.